### ISSUE TITLE
bug?(ios): unexpected error thrown if location services disabled

### ISSUE THREAD

**Issue by tafelnl** [ionic-team/capacitor-plugins#1035]
## Bug Report

I am reporting this as a bug, as it seems like unexpected behavior and is not documented anywhere. But please correct me if I am wrong (but that then probably means the docs should be updated).

### Plugin(s)
Geolcation

### Capacitor Version

```
  @capacitor/cli: 3.5.1
  @capacitor/core: 3.5.1
  @capacitor/android: 3.5.1
  @capacitor/ios: 3.5.1
```

### Platform(s)
ios

### Current Behavior
When location services are disabled, `checkPermissions` and `requestPermissions` throw the following error: `Location services are not enabled`

links to lines of code where error is thrown:
- [`checkPermissions`](https://github.com/ionic-team/capacitor-plugins/blob/7563040983ad397e28616246e7ed5ffce69727c2/geolocation/ios/Plugin/GeolocationPlugin.swift#L166)
- [`requestPermissions`](https://github.com/ionic-team/capacitor-plugins/blob/7563040983ad397e28616246e7ed5ffce69727c2/geolocation/ios/Plugin/GeolocationPlugin.swift#L194)

### Expected Behavior
- I couldn't find any documentation about this error. (Although when you dive in a bit more, it's pretty obvious why it happens). So likely many developers do not necessarily (explicitly) catch this error in their code. That can result in unexpected behavior. For example, we found out about this, after seeing many of these errors in our sentry dashboard.
- Also, on Android this error does not seem to be thrown in a similar situation (on Android the location permissions can be disabled as well).
- Besides, a similar issue (https://github.com/ionic-team/capacitor/issues/5615) was created for the Capacitor v2 plugin, but it was only labeled with `needs reproduction` and then closed after going stale.

The points above make it seem like it was indeed overlooked by and not intended behavior (again, correct me if I'm wrong).

I think it would make sense to either:
- not throw an error, but return an extra variable/enum type in the return object instead (that _could_ result in breaking behavior, so maybe it's not ideal)
- document the existence of this error (and maybe a simple example of what logical next steps could be in ones code)

### Code Reproduction

- Run a basic Capacitor app with Geolocation plugin installed on a real iOS device
- Disable location services
- Call either `checkPermissions` or `requestPermissions`

Basic reproduction: https://github.com/tafelnl/capacitor-reproduction-geolocation

**Comment by Ionitron** [comment]
This issue may need more information before it can be addressed. In particular, it will need a reliable Code Reproduction that demonstrates the issue.

Please see the Contributing Guide for [how to create a Code Reproduction](https://github.com/ionic-team/capacitor/blob/HEAD/CONTRIBUTING.md#creating-a-code-reproduction).

Thanks!
Ionitron 💙

**Comment by tafelnl** [comment]
@jcesarmobile Can you elaborate on what kind of reproduction you need? It happens on every iOS device with location services disabled AFAIK.

We have our sample rate set to 2% on Sentry (2% of errors are being sent), and it already tracked 630 events in the past 24 hours:
![image](https://user-images.githubusercontent.com/35837839/173317563-9d844e5c-a656-4381-8801-1ff315b2520b.png)

**Comment by jcesarmobile** [comment]
The link explains what a code reproduction is, a sample app that we can run with the minimal code required to reproduce the issue,

**Comment by tafelnl** [comment]
That seems a bit unnecessary as for how trivial this code is (it's just one line of code). But I'll see what I can do

**Comment by tafelnl** [comment]
Sorry for bumping the issue and going slightly bit off-topic:
Upon inspecting the docs, I found out that errors in general are not documented. So that makes this more of a documentation issue rather than a bug.

Still, besides that, I think it's weird that iOS and Android don't seem to have the same behavior when location permissions are disabled.

**Comment by jcesarmobile** [comment]
Thanks for the sample app.
I've been able to verify that iOS and Android behave differently when location services are disabled.
We feel like the right approach is iOS one, so we will update Android to behave the same way.

**Comment by tafelnl** [comment]
Thanks. Seems like the right way indeed. I know it's a little off-topic, but it's actually also quite closely related to this issue: are there any plans on documenting (and/or even adding Typings for) the possible errors thrown?

**Comment by jcesarmobile** [comment]
This has been fixed in 4.0.0 beta 0
